src/main/java/pl/umk/mat/git2befit/service/workout/TrainingPlanService.java

```diff
@@ -21,31 +24,53 @@
     private final TrainingPlanRepository trainingPlanRepository;
     private final UserRepository userRepository;
     private final ExerciseRepository exerciseRepository;
+    private final TrainingPlanManufacture manufacture;
 
     public TrainingPlanService(
             TrainingPlanRepository trainingPlanRepository,
             UserRepository userRepository,
-            ExerciseRepository exerciseRepository
+            ExerciseRepository exerciseRepository,
+            TrainingPlanManufacture manufacture
     ) {
         this.trainingPlanRepository = trainingPlanRepository;
         this.userRepository = userRepository;
         this.exerciseRepository = exerciseRepository;
+        this.manufacture = manufacture;
     }
 
-    public ResponseEntity<?> saveTrainingWithUserEmail(List<TrainingPlan> trainingPlans, String email) {
+    public ResponseEntity<?> generate(TrainingForm trainingForm, String authorizationToken) {
+        List<Training> trainingPlans;
+
         try {
-            Optional<User> user = userRepository.findByEmail(email);
-            if (user.isPresent()) {
-                trainingPlans.forEach(plan -> {
-                    plan.setUser(user.get());
-                    trainingPlanRepository.save(plan);
-                });
-                return ResponseEntity.ok().build();
-            } else {
-                return ResponseEntity.badRequest().header("Cause", "user not found").build();
-            }
-        } catch (Exception e) {
-            return ResponseEntity.badRequest().header("Cause", "cant be saved").build();
+            trainingPlans = manufacture.createTrainingPlan(trainingForm);
+        } catch (IllegalArgumentException e) {
+            return ResponseEntity.status(HttpStatus.CONFLICT).header("Cause", e.getMessage()).build();
+        }
+
+        TrainingPlan trainingPlan = new TrainingPlan(trainingForm, trainingPlans);
+
+        trainingPlan.setTitle(trainingForm.getTrainingType());
+        List<TrainingPlan> savedTrainingPlan = null;
+        if (authorizationToken != null) {
+            try {
+                String email = JWTService.parseEmail(authorizationToken);
+                savedTrainingPlan = saveTrainingWithUserEmail(List.of(trainingPlan), email);
+            } catch (Exception ignored) {}
+        }
+
+        return ResponseEntity.ok(savedTrainingPlan != null ? savedTrainingPlan.get(0) : trainingPlan);
```

Comment:
```suggestion
        Training coTuJest = savedTrainingPlan != null ? savedTrainingPlan.get(0) : trainingPlan;
         return coTuJest;
```

Reply:
Weź może to nazwij jakoś bo tak na pierwszy rzut to nie wiem o co chodzi i dlaczego tak